analyzer/block/block.go

```diff
@@ -50,6 +51,11 @@ type BlockProcessor interface {
 	// The method should be a no-op if all the blocks up to `height` have been analyzed
 	// in slow-sync mode, assuming no bugs in dead reckoning code.
 	FinalizeFastSync(ctx context.Context, lastFastSyncHeight int64) error
+	// QueueLength returns the number of known blocks that still need to be processed,
+	// calculated as the difference between the analyzer's highest processed block
+	// and the current block height on-chain (fetched by the node-stats analyzer).
+	// Note that the true count may be higher during fast-sync.
+	QueueLength(ctx context.Context, height uint64) (int, error)
```

Comment:
Does this really need to be implemented by the `processor`? The `blockBasedAnalyzer` knows the height that's being processed currently, and the node height is stored in the DB anyway.

Reply:
Yeah I was on the fence about this - I initially aimed to mirror item analyzers but I like your suggestion, it's cleaner. Will update